lib/game/game.dart

```diff
@@ -101,30 +107,52 @@ abstract class Game {
 
   /// Use this method to load the assets need for the game instance to run
   Future<void> onLoad() async {}
-
-  /// Returns the widget which will be show while the instance is loading
-  Widget loadingWidget() => Container();
 }
 
-class OverlayWidget {
-  final String name;
-  final Widget widget;
+/// A [ChangeNotifier] used to control the visibility of overlays on a [Game].
+///
+/// To learn more, see:
+/// - [HasWidgetsOverlay]
+/// - [GameWidget.withOverlay]
+class ActiveOverlaysNotifier extends ChangeNotifier {
+  final Set<String> _activeOverlays = {};
```

Comment:
What happens here if you have for example a dynamic amount of widgets, do you have to assign a unique string to each one of them then? There must be a better way to do this than to use strings?

Reply:
We have to apply that dynamic in a widget build context.  In order to have an active control of which overlay is active on the game side, we have to identify the overlays. 

>There must be a better way to do this than to use strings?
What is the problem with strings?

Reply:
I also see no problems with strings for keys

Reply:
strings are not type safe. but I think it would be overkill to do something else. like a typed enum, and dart enums aren't as powerful as java's to make this sort of thing easy. I am ok with the string as well

Reply:
we do have to throw a runtime error if it doesn't exist do the user can fix it.

Reply:
This can be solved with generics, but generics in dart are like all my cousins: ugly.

Reply:
Added runtime verification if the overlay exists.

Reply:
Just curious, how would you solve this with generics?

Reply:
The user would pass an enum via a generic and we would define this generic as the key type on our overlay builder map. 
But this would require a generic on the game class as well. Also, would be quite weird since we cant [constrain a generic to be an enum](https://stackoverflow.com/questions/59362403/dart-constrain-a-generic-type-t-to-an-enum).